lib/rules/declaration-block-no-shorthand-property-overrides/index.js

```diff
@@ -27,34 +27,36 @@ const rule = (primary) => {
 		}
 
 		eachDeclarationBlock(root, (eachDecl) => {
-			/** @type {Record<string, string>} */
-			const declarations = {};
+			/** @type {Map<string, string>} */
+			const declarations = new Map();
 
 			eachDecl((decl) => {
 				const prop = decl.prop;
 				const unprefixedProp = vendor.unprefixed(prop).toLowerCase();
 				const prefix = vendor.prefix(prop).toLowerCase();
 
-				const overrideables = /** @type {Record<string, Set<string>>} */ (shorthandData)[
-					unprefixedProp
-				];
+				const overrideables = /** @type {Map<string, Set<string>>} */ (
+					longhandSubPropertiesOfShorthandProperties
+				).get(unprefixedProp);
 
 				if (!overrideables) {
-					declarations[prop.toLowerCase()] = prop;
+					declarations.set(prop.toLowerCase(), prop);
 
 					return;
 				}
 
 				for (const longhandProp of overrideables) {
-					if (!Object.prototype.hasOwnProperty.call(declarations, prefix + longhandProp)) {
+					const declaration = declarations.get(prefix + longhandProp);
```

Comment:
[note] Refactor to use `Map` instead of `Object`; `hasOwnProperty` is no longer needed (safe access).

Reply:
Much nicer.